src/www/vpn_openvpn_server.php

```diff
@@ -202,6 +202,21 @@
 
         if ($result = openvpn_validate_cidr($pconfig['tunnel_network'], gettext('IPv4 Tunnel Network'), false, 'ipv4')) {
             $input_errors[] = $result;
+        } elseif (isset($pconfig['tunnel_network'])) {
+            // Check IPv4 tunnel_network pool size
+            list($ipv4tunnel_base, $ipv4tunnel_prefix) = explode('/',trim($pconfig['tunnel_network']));
+            if ($pconfig['dev_mode'] != "tap") {
```

Comment:
probably better to use `$pconfig['dev_mode'] == 'tun' `, future device modes (if they would exist) likely have other constraints.

Reply:
agree